src/js/actions/documents.js

```diff
@@ -856,12 +860,119 @@ define(function (require, exports) {
     beforeStartup.writes = [locks.JS_DOC];
 
     /**
-     * Initialize the inactive documents. (The active document is initialized beforeStartup.)
+     * Send info to search store about searching for documents and
+     * initialize the inactive documents. (The active document is initialized beforeStartup.)
      *
      * @param {{currentIndex: number, docCount: number}=} payload
      * @return {Promise}
      */
     var afterStartup = function (payload) {
+        var appStore = this.flux.store("application"),
+            docStore = this.flux.store("document");
+
+        /**
+         * Make list of currently open documents to be used as search options
+         * 
+         * @return {Immutable.List.<object>}
+        */
+        var currDocSearchOptions = function () {
+            var document = appStore.getCurrentDocument(),
+                openDocs = appStore.getOpenDocumentIDs().filterNot(function (doc) {
+                                return doc === document.id;
+                            }),
+                docMap = openDocs.map(function (doc) {
+                    return {
+                        id: "curr-doc_" + doc.toString(),
+                        title: docStore.getDocument(doc).name,
+                        type: "item",
+                        category: [strings.SEARCH.CATEGORIES.DOCUMENT, strings.SEARCH.CATEGORIES.CURRENT]
+                    };
+                }),
+                docLabel = {
```

Comment:
It seems like this logic here (of mapping an object to a search item) is being duplicated across different actions where you're creating search results. 

Instead, consider registering with a result type (to use with IDs), header title, category in addition to things you're already registering with. That way, this function would be a really simple return of `openDocs.map(doc => docStore.getDocument(doc).name)` mapping.